pkg/custompluginmonitor/custom_plugin_monitor.go

```diff
@@ -135,9 +136,58 @@ func (c *customPluginMonitor) generateStatus(result cpmtypes.Result) *types.Stat
 						result.Rule.Reason,
 						timestamp,
 					))
+
+					condition.Status = status
+					condition.Reason = result.Rule.Reason
+				} else if condition.Status != status && status == types.True {
+					// change 2: Condition status change from False/Unknown to True
+					condition.Transition = timestamp
+					var defaultConditionReason string
+					var defaultConditionMessage string
+					for j := range c.config.DefaultConditions {
+						defaultCondition := &c.config.DefaultConditions[j]
+						if defaultCondition.Type == result.Rule.Condition {
+							defaultConditionReason = defaultCondition.Reason
+							defaultConditionMessage = defaultCondition.Message
+							break
+						}
+					}
+
+					events = append(events, util.GenerateConditionChangeEvent(
+						condition.Type,
+						status,
+						defaultConditionReason,
+						timestamp,
+					))
+
+					condition.Status = status
+					condition.Message = defaultConditionMessage
+					condition.Reason = defaultConditionReason
+				} else if condition.Status != status {
+					// change 3: Condition status change from False to Unknown or vice versa
+					condition.Transition = timestamp
+					condition.Message = result.Message
+					events = append(events, util.GenerateConditionChangeEvent(
+						condition.Type,
+						status,
+						result.Rule.Reason,
+						timestamp,
+					))
+
+					condition.Status = status
+					condition.Reason = result.Rule.Reason
+				} else if condition.Status == status && condition.Message != result.Message {
+					// change 4: Condition status do not change. condition message changes.
+					condition.Transition = timestamp
+					condition.Message = result.Message
+					events = append(events, util.GenerateConditionChangeEvent(
+						condition.Type,
+						status,
+						condition.Reason,
+						timestamp,
+					))
 				}
-				condition.Status = status
```

Comment:
I feel like most code in the if-else clause can be extracted out.

At least event generation can be extracted.